### ISSUE TITLE
Transmission Fails to Start on DS414j after upgrading to DSM7 (GLIBC_2.25)

### ISSUE THREAD

**Issue by deano019s** [SynoCommunity/spksrc#5111]
### Is this a new Bug?

- [X] I checkd that the bug hasn't been reported before

### Package Name

Transmission

### Package Version

3.00-19

### Device Model

DS414j

### Device Architecture

~ARMv7~ comcerto2k

### Firmware Version

7.0.1-42218 Update 2

### What happened?

Trying to start transmission on DS414j with DSM7.0.1 results in the below:

Starting transmission command /volume1/@appstore/transmission/bin/transmission-daemon -g /volume1/@appdata/transmission -x /volume1/@appdata/transmission/transmission.pid -e /volume1/@appdata/transmission/transmission.log 
/volume1/@appstore/transmission/bin/transmission-daemon: /lib/libc.so.6: version `GLIBC_2.25' not found (required by /var/packages/transmission/target/lib/libevent-2.1.so.7)
/volume1/@appstore/transmission/bin/transmission-daemon: /lib/libc.so.6: version `GLIBC_2.25' not found (required by /var/packages/transmission/target/lib/libcrypto.so.1.1)
transmission is not running

### Reproduction steps

```bash
Start Transmission from GUI/CLI
```

### Install Log

```log
==> /var/log/packages/transmission.log <==
2022/01/30 12:04:45     ===> Step postupgrade. USER=sc-transmission GROUP=sc-download SHARE_PATH=/volume1/downloads
2022/01/30 12:04:45     Begin service_postupgrade
2022/01/30 12:04:45     /var/packages/transmission/scripts/service-setup: line 133: shopt: -d: invalid option
2022/01/30 12:04:45     shopt: usage: shopt [-pqsu] [-o] [optname ...]
2022/01/30 12:04:45     End service_postupgrade
2022/01/30 12:04:45     upgrade transmission 3.00-19 End postupgrade ret=[0]
2022/01/30 12:04:46     upgrade transmission 3.00-19 Begin start-stop-status start
2022/01/30 12:05:07     upgrade transmission 3.00-19 End start-stop-status start ret=[1]
2022/01/30 12:05:09     (system) trigger transmission 3.00-19 Begin start-stop-status stop
2022/01/30 12:05:09     (system) trigger transmission 3.00-19 End start-stop-status stop ret=[0]
```

### Service Log

```log
Starting transmission command /volume1/@appstore/transmission/bin/transmission-daemon -g /volume1/@appdata/transmission -x /volume1/@appdata/transmission/transmission.pid -e /volume1/@appdata/transmission/transmission.log 
/volume1/@appstore/transmission/bin/transmission-daemon: /lib/libc.so.6: version `GLIBC_2.25' not found (required by /var/packages/transmission/target/lib/libevent-2.1.so.7)
/volume1/@appstore/transmission/bin/transmission-daemon: /lib/libc.so.6: version `GLIBC_2.25' not found (required by /var/packages/transmission/target/lib/libcrypto.so.1.1)
transmission is not running
```

**Comment by BenjV** [comment]
The Linux on your Nas has a Glibc version lower then 2.25 on board and the current DSM 7 Transmission requires then 2.25 or higher.
You can check the version on your Nas with this command.
`/lib/libc.so.6`

**Comment by deano019s** [comment]
Ah right, you are spot on:

```
GNU C Library (crosstool-NG 1.20.0) stable release version 2.20-2014.11, by Roland McGrath et al.
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 4.9.3 20150311 (prerelease).
Available extensions:
        crypt add-on version 2.1 by Michael Glad and others
        GNU Libidn by Simon Josefsson
        Native POSIX Threads Library by Ulrich Drepper et al
        BIND-8.2.3-T5B
libc ABIs: UNIQUE
For bug reporting instructions, please see:
<http://www.gnu.org/software/libc/bugs.html>.
```

Not much I can do then aside from logging a case with Synology?

**Comment by BenjV** [comment]
You can use DownloadStation instead.
That is under the hood also Transmission.

**Comment by Hugues1337** [comment]
I just created a ticket at synology to update this package... no reason for this model to have a different version than others in DSM7. Tell me good luck.

**Comment by deano019s** [comment]
> I just created a ticket at synology to update this package... no reason for this model to have a different version than others in DSM7. Tell me good luck.

I did the same a while back, after Synology reviewed my NAS and logs, they provided the below:

> After confirming by the developers, due to the limited resource, we didn't update the tool (Glibc) on DS414j so you would encounter this issue. In addition, Transmission is not in the [supported list](https://kb.synology.com/DSM/tutorial/supported_third_party_packages_beta) so there is no plan to fix this issue at this moment. 
Therefore, we recommend you contact the developers of Transmission if it is possible to provide a special version for the older Glibc versions.

I'm guessing they will respond the same, but let us know regardless.

**Comment by hgy59** [comment]
This issue is solved with #4880, but transmission packages for DSM7 are not splitted for armv7 and comcerto2k yet.

Anyone can build transmission in spk/transmission folder with  `make arch-comcerto2k-7.0` - that should be running on DS414j.

**Comment by Hugues1337** [comment]
> 

they answered me this:
`Many thanks for your feedback. We will log in this feature and forward that to our development team. At the same time, it will be helpful to us to learn why you would need glibc 2.25? Is there any specific feature that is required? Many thanks for your feedback. Please do not hesitate to let me know if you have any questions. Thank you. `

Stay tuned.

**Comment by Hugues1337** [comment]
> This issue is solved with #4880, but transmission packages for DSM7 are not splitted for armv7 and comcerto2k yet.

I will try this and if I succeed I will make an addition in the README.

**Comment by deano019s** [comment]
> This issue is solved with #4880, but transmission packages for DSM7 are not splitted for armv7 and comcerto2k yet.
> 
> Anyone can build transmission in spk/transmission folder with `make arch-comcerto2k-7.0` - that should be running on DS414j.

Thanks for the advice, confirming I was able to build and reinstall transmission on the 414j running DSM7 with the above.
